docs/upgrade/upgrade-opensource/upgrade-guide-from-5.4-to-6.0/enable-consistent-topology.rst

```diff
@@ -3,4 +3,102 @@ Enable Consistent Topology Updates
 =====================================
 
 This article explains how to enable consistent topology changes
-when you upgrade from version 5.4 to 6.0.
\ No newline at end of file
+when you upgrade from version 5.4 to 6.0.
+
+Introduction
+============
+
+ScyllaDB Open Source 6.0 introduces :ref:`consistent topology changes based on Raft <raft-topology-changes>`.
+It is enabled automatically for newly created clusters, but - unlike in the case of schema managed on Raft -
+consistent topology changes are *not* automatically enabled for clusters that were upgraded from an older version.
+In those clusters, consistent topology changes need to be enabled with a dedicated upgrade procedure.
+
+Before running the procedure, you **must** check that the cluster meets some prerequisites
+and you **must** ensure that some administrative procedures will not be run
+while the upgrade procedure is in progress.
+
+.. _enable-raft-topology-6.0-prerequisites:
+
+Prerequisites
+=============
+
+* Make sure that all nodes in the cluster are upgraded to ScyllaDB Open Source 6.0.
+* Verify that :ref:`schema on raft is enabled <schema-on-raft-enabled>`.
+* Make sure that all nodes enabled ``SUPPORTS_CONSISTENT_TOPOLOGY_CHANGES`` cluster feature.
+  One way to verify this is to look for the following message in the log:
+
+  .. code-block:: none
+
+    features - Feature SUPPORTS_CONSISTENT_TOPOLOGY_CHANGES is enabled
+
+  Alternatively, this can be checked programmatically by checking whether ``value`` column under the key ``enabled_features`` contains the name of the feature in the ``system.scylla_local`` table. This can be done with the following query:
+
+  .. code-block:: none
+
+    select value from system.scylla_local where key = 'enabled_features';
+
+  It returns a string which contains a comma-separated list of names of the features that the node sees as enabled. You should wait until it contains ``SUPPORTS_CONSISTENT_TOPOLOGY_CHANGES``.
+
+* Make sure that all nodes are alive for the duration of the upgrade.
+
+.. _enable-raft-topology-6.0-forbidden-operations:
+
+Administrative operations which must not be running during upgrade
+==================================================================
+
+Make sure that administrative operations will not be running while upgrade is in progress.
+In particular, you must abstain from:
+
+* :doc:`Cluster management procedures </operating-scylla/procedures/cluster-management/index>` (adding, replacing, removing, decomissioning nodes etc.).
+* Runing :doc:`nodetool repair </operating-scylla/nodetool-commands/repair>`.
+* Runing :doc:`nodetool checkAndRepairCdcStreams </operating-scylla/nodetool-commands/checkandrepaircdcstreams>`.
+
```

Comment:
I got conditioned to ignore it because it always shows some issues outside of my code...

Reply:
Actually, I didn't notice because I didn't look at the commit view / files changed view (I didn't really have a reason to do so). IMO scylla-promoter should indicate in its report that there are some new types that need to be looked at.